main.go

```diff
@@ -57,3 +45,47 @@ func main() {
 
 	ctrl.Wait()
 }
+
+func newSyncConsumer(cfg *MateConfig) (consumers.Consumer, error) {
+	// New returns a Consumer implementation.
+	var consumer consumers.Consumer
+	var err error
+	switch cfg.Consumer {
+	case "google":
+		consumer, err = consumers.NewGoogleDNS(cfg.GoogleZone, cfg.GoogleProject, cfg.GoogleRecordGroupID)
+	case "aws":
+		consumer, err = consumers.NewAWSConsumer(cfg.AWSRecordGroupID)
+	case "stdout":
+		consumer, err = consumers.NewStdout()
+	default:
+		return nil, fmt.Errorf("Unknown consumer '%s'.", cfg.Consumer)
+	}
+	if err != nil {
+		return nil, err
+	}
+	return consumers.NewSynced(consumer)
+}
+
+func newProducer(cfg *MateConfig) (producers.Producer, error) {
+	switch cfg.Producer {
+	case "kubernetes":
+		kubeConfig := &producers.KubernetesOptions{
+			Format:    cfg.KubeFormat,
+			APIServer: cfg.KubeServer,
+		}
+		return producers.NewKubernetes(kubeConfig)
+	case "fake":
+		fakeConfig := &producers.FakeProducerOptions{
+			DNSName:       cfg.FakeDNSName,
+			FixedDNSName:  cfg.FakeFixedDNSName,
+			FixedHostname: cfg.FakeFixedHostname,
+			FixedIP:       cfg.FakeFixedIP,
+			Mode:          cfg.FakeMode,
+			TargetDomain:  cfg.FakeTargetDomain,
+		}
+		return producers.NewFake(fakeConfig)
+	case "null":
+		return producers.NewNull()
```

Comment:
let's change this to
* `producers.NewKubernetesProducer`
* `producers.NewFakeProducer`
* `producers.NewNullProducer`